### PR TITLE
update service account templates

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

This PR adds support for default service account for 

- alertmanager
- astro-ui
- cli-install
- kibana
- prometheus-blackbox-exporter

Additionally extends supports for existing 

- elastic services for es-client
- es-curator
- es-exporter
- es-nginx

## Related Issues

- https://github.com/astronomer/issues/issues/6747

## Testing

QA should see serviceAccountName configuration for above mapped services

## Merging

cherry-pick to 0.36.1

### CODE REVIEW

**Comment by Shubham-astro** [comment]
@pgvishnuram, The changes look good to me, however, I would suggest adding the new SA components in this test scripts: tests/chart_tests/test_byo_sa.py since we are introducing this capability to these components.

File: charts/astronomer/templates/_helpers.yaml [hunk 1]
```diff
@@ -135,6 +135,22 @@
 {{- end }}
 {{- end }}
 
+{{ define "astroUI.ServiceAccount" -}}
+{{- if and .Values.astroUI.serviceAccount.create .Values.global.rbacEnabled -}}
+{{ default (printf "%s-astro-ui" .Release.Name ) .Values.astroUI.serviceAccount.name }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/_helpers.yaml:140]:
```suggestion
{{ default (printf "%s-astro-ui" .Release.Name) .Values.astroUI.serviceAccount.name }}
```

File: charts/astronomer/templates/_helpers.yaml [hunk 2]
```diff
@@ -135,6 +135,22 @@
...
+{{- else -}}
+    {{ default "default" .Values.astroUI.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{ define "install.ServiceAccount" -}}
+{{- if and .Values.install.serviceAccount.create .Values.global.rbacEnabled -}}
+{{ default (printf "%s-cli-install" .Release.Name ) .Values.install.serviceAccount.name }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/_helpers.yaml:148]:
```suggestion
{{ default (printf "%s-cli-install" .Release.Name) .Values.install.serviceAccount.name }}
```

File: tests/chart_tests/test_byo_sa.py [hunk 3]
```diff
@@ -43,12 +48,26 @@ def test_serviceaccount_with_overrides(self, kube_version):
                 "charts/astronomer/templates/registry/registry-serviceaccount.yaml",
                 "charts/astronomer/templates/config-syncer/config-syncer-serviceaccount.yaml",
                 "charts/astronomer/templates/houston/api/houston-bootstrap-serviceaccount.yaml",
+                "charts/astronomer/templates/astro-ui/astro-ui-serviceaccount.yaml",
```
Comment (danielhoherd) [inline tests/chart_tests/test_byo_sa.py:51]: Can we change this list to a function that finds all `*serviceaccount*` templates? Like: 

```
>>> pp(sorted([str(x) for x in Path(git_root).glob('charts/**/*serviceaccount*.yaml')]))
[   'charts/astronomer/templates/commander/commander-serviceaccount.yaml',
    'charts/astronomer/templates/config-syncer/config-syncer-serviceaccount.yaml',
    'charts/astronomer/templates/houston/api/houston-bootstrap-serviceaccount.yaml',
    'charts/astronomer/templates/registry/registry-serviceaccount.yaml',
    'charts/elasticsearch/templates/es-serviceaccount.yaml',
    'charts/external-es-proxy/templates/external-es-proxy-serviceaccount.yaml',
    'charts/fluentd/templates/fluentd-serviceaccount.yaml',
    'charts/grafana/templates/grafana-bootstrap-serviceaccount.yaml',
    'charts/kube-state/templates/kube-state-serviceaccount.yaml',
    'charts/nginx/templates/nginx-default-backend-serviceaccount.yaml',
    'charts/nginx/templates/nginx-serviceaccount.yaml',
    'charts/pgbouncer/templates/pgbouncer-serviceaccount.yaml',
    'charts/postgresql/templates/serviceaccount.yaml',
    'charts/prometheus-node-exporter/templates/serviceaccount.yaml',
    'charts/prometheus-postgres-exporter/templates/serviceaccount.yaml',
    'charts/prometheus/templates/prometheus-serviceaccount.yaml']
```

**Review by danielhoherd — APPROVED** [review]
Looks good. I had one question about a way to improve the tests by discovering templates to test, but that's purely optional.

File: tests/chart_tests/test_byo_sa.py (hunk 3 above)
Comment (pgvishnuram) [reply tests/chart_tests/test_byo_sa.py:51]: okay let me include this change similar to what we are doing for other tests

**Pull request merged**